### PR TITLE
Aspnet core windsor final WithCrossWired() for fluent registrations

### PR DESCRIPTION
**Pull request by generik0**
Hi @fir3pho3nixx 

Any igood ideas to solving this problem for fluent registrations?

### CODE REVIEW

File: src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs [hunk 1]
```diff
@@ -42,6 +44,13 @@ private static void RegisterApplicationComponents(WindsorContainer container, IS
 		{
 			container.Register(Component.For<AnyService>().CrossWired(container, serviceCollection, LifestyleType.Singleton));
 			container.Register(Component.For<IHttpContextAccessor>().ImplementedBy<HttpContextAccessor>().CrossWired(container, serviceCollection, LifestyleType.Singleton));
+
+			container.Register(Classes.FromAssemblyInThisApplication(Assembly.GetCallingAssembly())
+				.Where(t => t.Name.StartsWith(Guid.NewGuid().ToString())) //dummy
+				.WithServiceAllInterfaces()
+				.WithServiceAllInterfaces()
+				.WithCrossWired()
+				.LifestyleTransient());
```
Comment (fir3pho3nixx) [inline src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: Yes, I can see what you are aiming for here. Try this:

```csharp
			container.Register(Classes.FromAssemblyInThisApplication(Assembly.GetCallingAssembly())
				.Where(t => t.Name.StartsWith(Guid.NewGuid().ToString())) //dummy
				.WithServiceAllInterfaces()
				.WithServiceAllInterfaces()
				.Configure(x => x.CrossWired(container, serviceCollection, LifestyleType.Transient)));

```

Comment (fir3pho3nixx) [reply src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: I am going to quickly add something to make this nicer, one sec.

Comment (fir3pho3nixx) [reply src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: You can now cross wire using this: 

```csharp
container.Register(Classes.FromAssemblyInThisApplication(Assembly.GetCallingAssembly())
				.Where(t => t.Name.StartsWith(Guid.NewGuid().ToString())) //dummy
				.WithServiceAllInterfaces()
				.WithServiceAllInterfaces()
				.CrossWired(container, serviceCollection, LifestyleType.Transient));
```

Comment (fir3pho3nixx) [reply src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: https://github.com/fir3pho3nixx/Windsor/commit/c4f337811769438159a3221f6500ff756fa3113a

**Comment by generik0** [comment]
@fir3pho3nixx when we register the fluent we also add a lifetime. Any way to grab that lifetime from the fluent registration?

**Comment by fir3pho3nixx** [comment]
@generik0 - Just checked this out and it is more difficult than one would expect. 

The problem trying so far is this: 

Descriptors are not publicly accessible from the ComponentRegistration object: https://github.com/castleproject/Windsor/blob/master/src/Castle.Windsor/MicroKernel/Registration/ComponentRegistration.cs#L48

They contain the knowledge of what happens when the LifestyleGroup is added to the ComponentRegistration using the AddDescriptor method. 

This also imposes an ordering problem for fluent registrations, LifestyleGroup has to be there for a lifestyle to be found. 

The easiest I can do is default the extension to make the third parameter optional and default it to scoped. Do you think that would work?

**Comment by generik0** [comment]
@fir3pho3nixx Well i think what you have done is already very good. I just think there seams to be a lot of double information. e.g. the container registers, but you ned to pass into the descriptor.
What ever your can do would be great. But the main focus is if it works or not.. And it will with proposal for default scoped in lifestyle

**Comment by fir3pho3nixx** [comment]
@generik0 

I will default them all to scoped then. 

Just want to thank you for getting involved with this. I think we have a much richer support for MVC because of your involvement. I am really satisfied that we have a hardened production ready facility here. I am going to credit you properly in the docs and I think we are ready to raise the PR back to the Castle Project. 

Please setup notifications for the Castle Windsor project so you can help out with any issues around this facility.

**Pull request closed without merging**